File: telegram-service/src/main/java/de/nak/home_assistant/commands/Start.java
package de.nak.home_assistant.commands;

import com.pengrad.telegrambot.TelegramBot;
import com.pengrad.telegrambot.model.Update;
import de.nak.home_assistant.CustomKeyboard;
import de.nak.home_assistant.models.telegram.Command;
import de.nak.home_assistant.services.database.UserService;

public class Start extends Command {

    private final TelegramBot bot;

    public Start(TelegramBot bot) {
        this.bot = bot;
        this.name = "start";
        this.description = "Startet den Bot";
    }

    public Start() {
        this(null);
    }

    public void reply(Update update) {
        long userId = update.message().from().id();
        long chatId = update.message().chat().id();

        UserService userService = new UserService(userId);
        boolean success = userService.createUser();

        String text;
        if (success){
            text = "🚀 Willkommen beim Home Assistant Bot" +
                    "\n\n⚙️ Als nächstes kannst du mit dem '" + new Settings().getName() + "' Kommando deine Module verwalten." +
                    "\n🏁 Wenn du schon Module hinzugfügt hast kannst du auch direkt loslegen in dem du diese im Menü auswählst";
        } else {
            text = "Fehler beim Anmelden. Versuch es später erneut. :)";
        }

        bot.execute(CustomKeyboard.generateMessageWithKeyboard(userId, chatId, text));
    }
}

File: telegram-service/src/commands/start.ts
import { Message } from "node-telegram-bot-api";
import { Command } from "@telegram-home-assistant/types";
import { bot } from "../bot";
import { i18n } from "../i18n";
import { generateSendMessageOptions } from "../keyboard";
import { createUser } from "../services/user";

export const Start: Command = () => {
  const translations = i18n();

  console.log(translations);
  return {
    command: translations.start.name,
    description: translations.start.description,
  };
};

export const Back: Command = () => {
  const translations = i18n();

  return {
    command: translations.back.name,
    description: translations.back.description,
  };
};

export const replyToStart = async ({ chat, from }: Message) => {
  const success = await createUser(from.id);
  const translations = i18n();

  const text = success ? translations.start.message : translations.start.error;

  return generateSendMessageOptions(from.id)
    .then((options) => bot.sendMessage(chat.id, text, options))
    .catch(() => bot.sendMessage(chat.id, text));
};

export const replyToBack = async ({ chat, from }: Message) => {
  const success = await createUser(from.id);
  const translations = i18n();

  const text = success ? translations.back.success : translations.back.error;

  return generateSendMessageOptions(from.id)
    .then((options) => bot.sendMessage(chat.id, text, options))
    .catch(() => bot.sendMessage(chat.id, text));
};
